samples/wifi/shell/src/wifi_cred_shell.c

```diff
@@ -0,0 +1,229 @@
+		ssid_len, ssid, ssid_len,
+		wifi_security_txt(creds.header.type));
+
+
+	if (creds.header.type == WIFI_SECURITY_TYPE_PSK ||
+	    creds.header.type == WIFI_SECURITY_TYPE_PSK_SHA256 ||
+	    creds.header.type == WIFI_SECURITY_TYPE_SAE) {
+		printk(", password: \"%.*s\", password_len: %d",
+		       creds.password_len, creds.password, creds.password_len);
+	}
+
+	if (creds.header.flags & WIFI_CREDENTIALS_FLAG_BSSID) {
+		printk(", bssid: " MACSTR,
+		       creds.header.bssid[0], creds.header.bssid[1],
+		       creds.header.bssid[2], creds.header.bssid[3],
+		       creds.header.bssid[4], creds.header.bssid[5]);
+	}
+
+	if (creds.header.flags & WIFI_CREDENTIALS_FLAG_2_4GHz) {
+		printk(", band: 2.4GHz");
+	}
+
+	if (creds.header.flags & WIFI_CREDENTIALS_FLAG_5GHz) {
+		printk(", band: 5GHz");
+	}
+
+	if (creds.header.flags & WIFI_CREDENTIALS_FLAG_FAVORITE) {
+		printk(", favorite");
+	}
+	printk("\n");
+}
+
+static enum wifi_security_type parse_sec_type(const char *s)
+{
+	if (strcmp("OPEN", s) == 0) {
+		return WIFI_SECURITY_TYPE_NONE;
+	}
+	if (strcmp("WPA2-PSK", s) == 0) {
+		return WIFI_SECURITY_TYPE_PSK;
+	}
+	if (strcmp("WPA2-PSK-SHA256", s) == 0) {
+		return WIFI_SECURITY_TYPE_PSK_SHA256;
+	}
+	if (strcmp("WPA3-SAE", s) == 0) {
+		return WIFI_SECURITY_TYPE_SAE;
+	}
+	return WIFI_SECURITY_TYPE_UNKNOWN;
+}
+
+static enum wifi_frequency_bands parse_band(const char *s)
+{
+	if (strcmp("2.4GHz", s) == 0) {
+		return WIFI_FREQ_BAND_2_4_GHZ;
+	}
+	if (strcmp("5GHz", s) == 0) {
+		return WIFI_FREQ_BAND_5_GHZ;
+	}
+	if (strcmp("6GHz", s) == 0) {
+		return WIFI_FREQ_BAND_6_GHZ;
+	}
+	return WIFI_FREQ_BAND_UNKNOWN;
+}
+
+static int cmd_auto_connect(const struct shell *sh, size_t argc, char *argv[])
+{
+	struct net_if *iface = net_if_get_default();
+	int rc = net_mgmt(NET_REQUEST_WIFI_CONNECT_STORED, iface, NULL, 0);
+
+	if (rc) {
+		printk("an error occurred when trying to auto-connect to a network. err: %d", rc);
+	}
+	return 0;
+}
+
+static int cmd_add_network(const struct shell *sh, size_t argc, char *argv[])
+{
+	int ret;
+
+	if (argc < 3) {
+		goto help;
+	}
+
+	struct wifi_credentials_personal creds = {
+		.header.ssid_len = strlen(argv[1]),
+		.header.type = parse_sec_type(argv[2]),
+	};
+
+	memcpy(creds.header.ssid, argv[1], creds.header.ssid_len);
+
+	if (creds.header.type == WIFI_SECURITY_TYPE_UNKNOWN) {
+		printk("cannot parse security type!\n");
+		goto help;
+	}
+
+	size_t arg_idx = 3;
+
+	if (creds.header.type == WIFI_SECURITY_TYPE_PSK ||
+	    creds.header.type == WIFI_SECURITY_TYPE_PSK_SHA256 ||
+	    creds.header.type == WIFI_SECURITY_TYPE_SAE) {
+		/* parse passphrase */
+		if (argc < 4) {
+			printk("missing password\n");
+			goto help;
+		}
+		creds.password_len = strlen(argv[3]);
+		if (creds.password_len > CONFIG_WIFI_CREDENTIALS_SAE_PASSWORD_LENGTH ||
+			(
+				(creds.header.type == WIFI_SECURITY_TYPE_PSK ||
+				 creds.header.type == WIFI_SECURITY_TYPE_PSK_SHA256) &&
+				creds.password_len > WIFI_PSK_MAX_LEN
+			)
+		) {
```

Comment:
```suggestion
		creds.password_len = strnlen(argv[3], WIFI_CREDENTIALS_MAX_PASSWORD_LEN + 1); 
		if ((
				creds.header.type == WIFI_SECURITY_TYPE_SAE &&
				creds.password_len > CONFIG_WIFI_CREDENTIALS_SAE_PASSWORD_LENGTH
			)||(
				(creds.header.type == WIFI_SECURITY_TYPE_PSK ||
				 creds.header.type == WIFI_SECURITY_TYPE_PSK_SHA256) &&
				creds.password_len > WIFI_PSK_MAX_LEN
			)
		) {
```
Since CONFIG_WIFI_CREDENTIALS_SAE_PASSWORD_LENGTH and WIFI_PSK_MAX_LEN are (theoretically) independent, I think it makes sense to validate the lengths explicitly for each type like this or similar.

Reply:
applied the idea now, but slightly different